### PR TITLE
Initial Azure.ResourceManager.ContainerInstance

### LINKED ISSUE #28691: Initial Azure.ResourceManager.ContainerInstance

### PR DESCRIPTION
**Pull request by HarveyLink**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/Container.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using Azure.Core;
+
+namespace Azure.ResourceManager.ContainerInstance.Models
+{
+    /// <summary> A container instance. </summary>
+    public partial class Container
```
Comment (Yao725) [inline sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/Container.cs:15]: Single word class name, please rename it.

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/ContainerGroupResource.cs [hunk 2]
```diff
@@ -0,0 +1,800 @@
...
+        }
+
+        /// <summary>
+        /// Gets all the network dependencies for this container group to allow complete control of network setting and configuration. For container groups, this will always be an empty list.
+        /// Request Path: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerInstance/containerGroups/{containerGroupName}/outboundNetworkDependenciesEndpoints
+        /// Operation Id: ContainerGroups_GetOutboundNetworkDependenciesEndpoints
+        /// </summary>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <returns> An async collection of <see cref="string" /> that may take multiple service requests to iterate over. </returns>
+        public virtual AsyncPageable<string> GetOutboundNetworkDependenciesEndpointsAsync(CancellationToken cancellationToken = default)
```
Comment (Yao725) [inline sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/ContainerGroupResource.cs:403]: This return type really looks weird ==

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/Capabilities.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            Location = location;
+            IPAddressType = ipAddressType;
+            Gpu = gpu;
+            CapabilitiesValue = capabilitiesValue;
+        }
+
+        /// <summary> The OS type that this capability describes. </summary>
+        public string OSType { get; }
+        /// <summary> The resource location. </summary>
+        public string Location { get; }
```
Comment (Yao725) [inline sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/Capabilities.cs:43]: `AzureLocation`?

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/ContainerGroupNetworkProtocol.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            _value = value ?? throw new ArgumentNullException(nameof(value));
+        }
+
+        private const string TCPValue = "TCP";
+        private const string UDPValue = "UDP";
+
+        /// <summary> TCP. </summary>
+        public static ContainerGroupNetworkProtocol TCP { get; } = new ContainerGroupNetworkProtocol(TCPValue);
+        /// <summary> UDP. </summary>
+        public static ContainerGroupNetworkProtocol UDP { get; } = new ContainerGroupNetworkProtocol(UDPValue);
```
Comment (Yao725) [inline sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/ContainerGroupNetworkProtocol.cs:31]: `Tcp` ans `Udp`?

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/ContainerGroupSubnetId.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        /// <param name="id"> Resource ID of virtual network and subnet. </param>
+        /// <param name="name"> Friendly name for the subnet. </param>
+        internal ContainerGroupSubnetId(string id, string name)
+        {
+            Id = id;
+            Name = name;
+        }
+
+        /// <summary> Resource ID of virtual network and subnet. </summary>
+        public string Id { get; set; }
```
Comment (Yao725) [inline sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/ContainerGroupSubnetId.cs:38]: `ResourceIdentifier`?

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/Scheme.cs [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.ContainerInstance.Models
+{
+    /// <summary> The scheme. </summary>
+    public readonly partial struct Scheme : IEquatable<Scheme>
```
Comment (Yao725) [inline sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/Scheme.cs:14]: Single word.

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/ContainerNetworkProtocol.cs [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+            _value = value ?? throw new ArgumentNullException(nameof(value));
+        }
+
+        private const string TCPValue = "TCP";
+        private const string UDPValue = "UDP";
+
+        /// <summary> TCP. </summary>
+        public static ContainerNetworkProtocol TCP { get; } = new ContainerNetworkProtocol(TCPValue);
+        /// <summary> UDP. </summary>
+        public static ContainerNetworkProtocol UDP { get; } = new ContainerNetworkProtocol(UDPValue);
```
Comment (Yao725) [inline sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/ContainerNetworkProtocol.cs:31]: `Tcp` and `Udp`?

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/Event.cs [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+
+namespace Azure.ResourceManager.ContainerInstance.Models
+{
+    /// <summary> A container group or container instance event. </summary>
+    public partial class Event
```
Comment (Yao725) [inline sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/Event.cs:13]: Single word class name.

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/api/Azure.ResourceManager.ContainerInstance.netstandard2.0.cs [hunk 9]
```diff
@@ -0,0 +1,558 @@
...
+        public virtual Azure.Response Stop(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.Threading.Tasks.Task<Azure.Response> StopAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.Response<Azure.ResourceManager.ContainerInstance.ContainerGroupResource> Update(Azure.ResourceManager.ContainerInstance.Models.ContainerGroupPatch patch, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.Threading.Tasks.Task<Azure.Response<Azure.ResourceManager.ContainerInstance.ContainerGroupResource>> UpdateAsync(Azure.ResourceManager.ContainerInstance.Models.ContainerGroupPatch patch, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+    }
+    public static partial class ContainerInstanceExtensions
```
Comment (Yao725) [inline sdk/containerinstance/Azure.ResourceManager.ContainerInstance/api/Azure.ResourceManager.ContainerInstance.netstandard2.0.cs:71]: All the method names that have a `Locations` suffix don't look clear enough, maybe change them to `Get***WithLocation`?

**Review by Yao725 — COMMENTED** [review]
Left some comments, and by the way, you might need to update the cspell check file to fix the broken CI.

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/Container.cs (hunk 1 above)
Comment (HarveyLink) [reply sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/Container.cs]: Renamed with ContainerInstanceContainer

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/Volume.cs [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using Azure.Core;
+
+namespace Azure.ResourceManager.ContainerInstance.Models
+{
+    /// <summary> The properties of the volume. </summary>
+    public partial class Volume
```
[the remark this answers is not in the document]
Comment (HarveyLink) [reply sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/Volume.cs]: Renamed with ContainerInstanceVolume

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/ContainerAttachResponse.cs [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+
+namespace Azure.ResourceManager.ContainerInstance.Models
+{
+    /// <summary> The information for the output stream from container attach. </summary>
+    public partial class ContainerAttachResponse
```
[the remark this answers is not in the document]
Comment (HarveyLink) [reply sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/ContainerAttachResponse.cs]: Fixed

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/ContainerExecResponse.cs [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+
+namespace Azure.ResourceManager.ContainerInstance.Models
+{
+    /// <summary> The information for the container exec command. </summary>
+    public partial class ContainerExecResponse
```
[the remark this answers is not in the document]
Comment (HarveyLink) [reply sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/ContainerExecResponse.cs]: Fixed

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/Logs.cs [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+namespace Azure.ResourceManager.ContainerInstance.Models
+{
+    /// <summary> The logs. </summary>
+    public partial class Logs
```
[the remark this answers is not in the document]
Comment (HarveyLink) [reply sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/Logs.cs]: Renamed with ContainerLogs

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/ContainerGroupResource.cs (hunk 2 above)
Comment (HarveyLink) [reply sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/ContainerGroupResource.cs:403]: [The swagger definition](https://github.com/Azure/azure-rest-api-specs/blob/60cd4dbca5214118b763d1edc835a44c8386283a/specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2021-10-01/containerInstance.json#L2249):
    "NetworkDependenciesResponse": {
      "description": "Response for network dependencies, always empty list.",
      "type": "array",
      "items": {
        "type": "string"
      }
    }
I guess the return type is acceptable

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/Capabilities.cs [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+using Azure.Core;
+using Azure.ResourceManager.Models;
+
+namespace Azure.ResourceManager.ContainerInstance.Models
+{
+    /// <summary> The regional capabilities. </summary>
+    public partial class Capabilities : ResourceData
+    {
+        /// <summary> Initializes a new instance of Capabilities. </summary>
+        internal Capabilities()
```
[the remark this answers is not in the document]
Comment (HarveyLink) [reply sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/Capabilities.cs:14]: Renamed with ContainerInstanceCapabilities

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/Capabilities.cs (hunk 3 above)
Comment (HarveyLink) [reply sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/Capabilities.cs]: Update to AzureLocation

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/ContainerGroupNetworkProtocol.cs (hunk 4 above)
Comment (HarveyLink) [reply sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/ContainerGroupNetworkProtocol.cs]: Fixed

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/ContainerGroupSubnetId.cs (hunk 5 above)
Comment (HarveyLink) [reply sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/ContainerGroupSubnetId.cs]: Updated to ResourceIdentifier

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/Scheme.cs (hunk 6 above)
Comment (HarveyLink) [reply sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/Scheme.cs:14]: Renamed with ContainerInstanceScheme

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/ContainerNetworkProtocol.cs (hunk 7 above)
Comment (HarveyLink) [reply sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/ContainerNetworkProtocol.cs]: Fixed

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/InitContainerDefinition.cs [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using Azure.Core;
+
+namespace Azure.ResourceManager.ContainerInstance.Models
+{
+    /// <summary> The init container definition. </summary>
+    public partial class InitContainerDefinition
```
[the remark this answers is not in the document]
Comment (HarveyLink) [reply sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/InitContainerDefinition.cs]: Renamed with InitContainerDefinitionContent

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/LogAnalytics.cs [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+        {
+            WorkspaceId = workspaceId;
+            WorkspaceKey = workspaceKey;
+            LogType = logType;
+            Metadata = metadata;
+            WorkspaceResourceId = workspaceResourceId;
+        }
+
+        /// <summary> The workspace id for log analytics. </summary>
+        public string WorkspaceId { get; set; }
```
[the remark this answers is not in the document]
Comment (HarveyLink) [reply sdk/containerinstance/Azure.ResourceManager.ContainerInstance/src/Generated/Models/LogAnalytics.cs]: Updated to ResourceIdentifier

File: sdk/containerinstance/Azure.ResourceManager.ContainerInstance/api/Azure.ResourceManager.ContainerInstance.netstandard2.0.cs (hunk 9 above)
Comment (HarveyLink) [reply sdk/containerinstance/Azure.ResourceManager.ContainerInstance/api/Azure.ResourceManager.ContainerInstance.netstandard2.0.cs:71]: Fixed

**Review by Yao725 — APPROVED** [review]
In general, LGTM

**Pull request merged**